apps/prairielearn/src/components/QuestionScore.html.ts

```diff
@@ -454,3 +463,93 @@ export function ExamQuestionAvailablePoints({
     </a>
   `;
 }
+
+function QuestionValue({
+  instance_question,
+  assessment_question,
+}: {
+  instance_question: InstanceQuestion;
+  assessment_question: AssessmentQuestion;
+}) {
+  const initAutoPoints =
+    (assessment_question.init_points ?? 0) - (assessment_question.max_manual_points ?? 0);
+
+  const currentAutoValue =
+    (instance_question.current_value ?? 0) - (assessment_question.max_manual_points ?? 0);
+
+  const bestCurrentScore = run(() => {
+    const variantPoints = instance_question.variants_points_list.at(-1);
+    if (variantPoints == null) return 0;
+    if (variantPoints < initAutoPoints) return (variantPoints / initAutoPoints) * 100;
+    return 0;
+  });
+
+  const popoverContent = run(() => {
+    const parts: HtmlValue[] = [
+      'This question awards partial credit if you continue getting closer to the correct answer.',
+    ];
+
+    if (bestCurrentScore === 0) {
+      parts.push(html`<hr />`);
+      parts.push(
+        `If you score 100% on your next submission, you will be awarded an additional ${formatPoints(currentAutoValue)} points.`,
+      );
+
+      parts.push(html`<hr />`);
+      parts.push(
+        html`If you score less than 100%, you will be awarded an additional
+          <code>${formatPoints(initAutoPoints)} * score / 100</code> points.`,
+      );
+    } else {
+      parts.push(html`<hr />`);
+
+      if (instance_question.some_perfect_submission) {
+        parts.push(
+          `Your highest submission score since your last 100% submission is ${formatPoints(bestCurrentScore)}%.`,
+        );
+      } else {
+        parts.push(`Your highest submission score so far is ${formatPoints(bestCurrentScore)}%.`);
+      }
+
+      // TODO: do we want to show the full formula, or just the number of points?
+      const perfectAdditionalPoints = (currentAutoValue * (100 - bestCurrentScore)) / 100;
+      parts.push(html`<hr />`);
+      parts.push(
+        `If you score 100% on your next submission, you will be awarded an additional ${formatPoints(perfectAdditionalPoints)} points.`,
+      );
+
+      parts.push(html`<hr />`);
+      parts.push(
+        html`If you score between ${formatPoints(bestCurrentScore)}% and 100%, you will be awarded
+          an additional
+          <code>
+            ${formatPoints(currentAutoValue)} * (score - ${formatPoints(bestCurrentScore)}) / 100
+          </code>
+          points.`,
+      );
+      parts.push(html`<hr />`);
+      parts.push(html`If you score less than 100%, you will not be awarded any additional points.`);
+    }
+
+    return joinHtml(parts);
+  });
+
+  // TODO: do we want to show `current_value` or `currentAutoValue` to the student?
+  // We currently show `current_value` on the assessment instance page, but when manual
+  // points are involved, showing `current_value` could be confusing, as the student can
+  // only actually earn `currentAutoValue` points.
```

Comment:
We need to make a decision here. Whatever we do, we should be consistent between here and the assessment instance question table.

Reply:
This leads to some interesting situations:

<img width="1326" alt="Screenshot 2024-10-16 at 15 58 06" src="https://github.com/user-attachments/assets/b974db71-589b-499c-a1b3-e3d185e2f252">
<img width="323" alt="Screenshot 2024-10-16 at 15 57 53" src="https://github.com/user-attachments/assets/0c47b939-837c-49bf-b16e-532211169d04">

I'm wondering if we should make some changes for purely-manually-graded Homework questions:

- On the assessment instance page, show a `-`.
- On the instance question page, hide the "Value" row entirely.

Does this sound reasonable? Another approach would be to update the popover of the instance question page to show a special-cased message that calls out that it's manually graded.

Reply:
Discussed with @mwest1066, he's on board with the above. I'll need to check that I'm doing something sensible in the case of 0 auto _and_ manual points.

Reply:
I made this change in ea6c9f7. Discussion continued further down in the PR comments.